(GH-1835) Document plan_hierarchy usage (#2162)
puppetlabs/bolt: documentation/applying_manifest_blocks.md

encrypt your data, the gem dependencies must be available to Bolt. See [Install
gems with Bolt packages](bolt_installing.md#).

> [Hiera interpolations](https://puppet.com/docs/puppet/latest/hiera_merging.html#hiera_interpolation)
> are not supported in Bolt outside of apply blocks, since Hiera data is looked up per-target but
> plans do not run in a per-target context. If you want to use Hiera to look up data outside of apply
> block and have an existing Hiera config that contains interpolations, you can add the data you want
> to use outside of apply blocks to the top-level `plan_hierarchy` key. If present, the data under
> this key will be used outside of apply blocks, while the typical hierarchy key will be used inside
> apply blocks. This allows users to use their existing Hiera configs in Bolt plans without erroring
> if any interpolation exists and they look up data outside an apply block.

Reviewer: ```suggestion
[Hiera interpolations](https://puppet.com/docs/puppet/latest/hiera_merging.html#hiera_interpolation)
are not supported in Bolt outside of apply blocks, because Hiera data is looked up per-target and
plans do not run in a per-target context. If you want to use Hiera to look up data outside of an apply
block and have an existing Hiera config that contains interpolations, you can add the data you want
to use outside of the apply block to the top-level `plan_hierarchy` key. If present, Bolt uses the data 
under the `plan_hierarchy` key outside of apply blocks, and uses the the typical hierarchy key inside 
of apply blocks. This allows you to use your existing Hiera configs in Bolt plans without encountering 
an error if interpolation exists and your plan tries to look up data outside of an apply block.
```

For example, with this Hiera configuration at `<project>/hiera.yaml`:
```yaml
version: 5

hierarchy:
  - name: "Per-OS defaults"
    path: "os/%{facts.os.family}.yaml"
  - name: Common
    path: hierarchy.yaml

plan_hierarchy:
  - name: Common
    path: plan_hierarchy.yaml
```

> The `plan_hierarchy` data will be used for the first `lookup()` call, and the regular `hierarchy`
> data will be used for the second `lookup()` call:

Reviewer: ```suggestion
Bolt uses the `plan_hierarchy` data for the first `lookup()` call, and the regular `hierarchy`
data for the second `lookup()` call:
```
